Modules/atexitmodule.c

```diff
@@ -38,41 +50,49 @@ PyUnstable_AtExit(PyInterpreterState *interp,
     callback->next = NULL;
 
     struct atexit_state *state = &interp->atexit;
+    _PyAtExit_LOCK(state);
     if (state->ll_callbacks == NULL) {
         state->ll_callbacks = callback;
         state->last_ll_callback = callback;
     }
     else {
         state->last_ll_callback->next = callback;
```

Comment:
Is this code correct? Don't we have to call `state->last_ll_callback = callback;` after?